### PR TITLE
telemetry: use Sentry tags instead of sentry context so they can be searched

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: src/ui/components/shared/LaunchButton.tsx [hunk 1]
```diff
@@ -7,7 +7,10 @@ import { UIState } from "ui/state";
 const { features } = require("ui/utils/prefs");
 
 function ShareButton({ setModal, recordingId }: PropsFromRedux) {
-  const onClick = () => setModal("browser-launch");
+  const onClick = () => {
+    console.log("hi");
```
Comment (jaril) [inline src/ui/components/shared/LaunchButton.tsx]: lol `console.log`

Comment (jaril) [reply src/ui/components/shared/LaunchButton.tsx]: well that and the rest of it really

Comment (jazzdan) [inline src/ui/components/shared/LaunchButton.tsx:11]: hehe ignore this

**Review by jaril — APPROVED** [review]
Looks good minus the changes in launchbutton.tsx

File: src/ui/utils/telemetry.ts [hunk 2]
```diff
@@ -39,18 +39,21 @@ export function setTelemetryContext(
   userEmail: string | undefined,
   isInternal: boolean
 ) {
-  let sentryContext: Record<string, string | boolean> = { isInternal };
+  Sentry.setTag("internal", isInternal);
+  if (userId && userEmail) {
+    Sentry.setUser({ id: userId, email: userEmail });
+    Sentry.setTag("userEmail", userEmail);
+  } else {
+    Sentry.setTag("anonymous", true);
```
Comment (loganfsmyth) [inline src/ui/utils/telemetry.ts:47]: Would it be worth making this more consistent by including 
```
Sentry.setTag("anonymous", false);
```
in the first `if` block too?

**Review by loganfsmyth — APPROVED** [review]

File: src/ui/utils/telemetry.ts [hunk 3]
```diff
@@ -39,18 +39,21 @@ export function setTelemetryContext(
   userEmail: string | undefined,
   isInternal: boolean
 ) {
-  let sentryContext: Record<string, string | boolean> = { isInternal };
+  Sentry.setTag("internal", isInternal);
+  if (userId && userEmail) {
+    Sentry.setUser({ id: userId, email: userEmail });
+    Sentry.setTag("userEmail", userEmail);
```
[the remark this answers is not in the document]
Comment (jazzdan) [reply src/ui/utils/telemetry.ts:45]: I _think_ so. Here's what I saw when I didn't set userEmail as a separate tag:

Screen Shot 2021-06-30 at 2 56 51 PM

Then user email was down there in context metadata on the bottom:

Screen Shot 2021-06-30 at 2 57 04 PM

But, maybe you can use that `:` syntax in the first image to search inside the user object? It seemed more foolproof to do it this way, heh

[the remark this answers is not in the document]
Comment (loganfsmyth) [reply src/ui/utils/telemetry.ts:45]: Sounds good!

**Pull request merged**